Updated compatibility to PHP 7.4

RSalo opened the pull request:
1) Updated unit-test to compatibility PHP 7.4 and PHPUnit 8.5
2) Updated namespaces to match PSRs'
3) Small code clean up

| Q             | A
| ------------- | ---
| Branch?       | "master"
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | yes
| License       | MIT

RSalo committed:
1) Updated unit-test to compatibility PHP 7.4 and PHPUnit 8.5
2) Updated namespaces to match PSRs'
3) Small code clean up

RSalo commented:
@dbu can we add a branch for possible BC failures?
This travis failed due to hardcoded SYMFONY_PHPUNIT_VERSION in .travis,yml(not sure). If you remove this global variable, this will be detected automatically by your PHP version. Also, I added PR to update SYMFONY_PHPUNIT_VERSION for the latest version in symfony phpunit-bridge package.
And can you add in travis PHP 7.4 and update symfony version to 4.4 and 5.1? Or can I do it also?

dbu commented:
thanks!

we set the phpunit version to avoid random build break because suddenly we would run a new major phpunit version. you can edit `.travis.yml` to change it to the right version. that can also be done per build target if necessary (on the `env:` line). i am glad if you can add 7.3 and 7.4 in the .travis.yml file to see if things work.

RSalo committed:
Change Travis configs

RSalo commented:
@dbu check this out https://travis-ci.com/github/RSalo/resource-rest-bundle We have to increase minimal PHP version to 7.2+. Is this will be good for the next major version? And yet. I see in .travis file this sentence "It's auto-generated by symfony-cmf/dev-kit package". If I correct understanding, should we generate DevOps files with symfony-cmf/dev-kit or sonata-project/dev-kit? But I don't see that one in our composer.json.

dbu commented:
dropping php 7.1 is fine.

dropping 7.1 is no BC break, because composer simply won't install the next version on php 7.1 but stick with the previous one. we should however do a new minor version, so that we can do important security fixes for 7.1 if really needed.

the auto-generated warning is no longer true actually, i don't use that dev-kit. please delete the line when you update the file.

RSalo committed:
Remove 7.1 from travis and increased to 7.2 in composer

RSalo committed:
Remove 7.1 from travis and increased to 7.2 in composer

RSalo committed:
Remove 7.1 from travis and increased to 7.2 in composer

RSalo commented:
Done. We should add a new branch 1.2-dev or something like that for now, add a milestone or tag for version 1.2, and add to there that kind of issue/pr like this. And remain 1.0 and 1.1 only for minor security fixes.

RSalo committed:
Add example versions to readme file

dbu commented on .travis.yml:
```diff
@@ -26,33 +22,21 @@ cache:
-    - php: 7.2
-      env: STABILITY=dev SYMFONY_VERSION=4.0.*
-    - php: 7.1
-      env: STABILITY=dev COMPOSER_FLAGS="--prefer-lowest" SYMFONY_VERSION=2.8.* SYMFONY_DEPRECATIONS_HELPER=weak
```
please keep the lowest version build, but with php 7.2 

we should be able to remove the SYMFONY_VERSION constant from the line though

RSalo replied:
I don't get it. What do you mean about the lowest version build? I did the lowest version 7.2. Or is there something else?

dbu replied:
this build sets `COMPOSER_FLAGS="--prefer-lowest"` to have composer install the lowest possible version of everything. this is to detect when we accidentally start using something that is only available in newer versions of our dependencies. then we can either not use it, or raise the minimum version.

the flag only makes sense in reusable libraries like this one. for applications on the other hand, i never bother with that, as we will install the newest possible version of everything anyways.

dbu commented on .travis.yml:
```diff
@@ -26,33 +22,21 @@ cache:
-      env: STABILITY=dev SYMFONY_VERSION=3.3.*
-    - php: 7.2
-      env: STABILITY=dev SYMFONY_VERSION=3.4.*
-    - env: TEST_INSTALLATION=true
```
please keep the test installation build

dbu commented on README.md:
```diff
@@ -30,11 +30,16 @@ KERNEL_CLASS="Symfony\Cmf\Bundle\ResourceRestBundle\Tests\Fixtures\App\Kernel" .
 
 ## Requirements
-
+Version <= 1.1:
```
lets just remove here everything except the line 
```
 See the `require` section of [composer.json](composer.json)
```

in line with other cmf bundles: https://github.com/symfony-cmf/routing-bundle/blob/master/README.md

dbu reviewed (commented):
thanks again! for the branches, we tag from the `master` branch. if you can add an entry to the CHANGELOG.md file titled `1.2.0` with a note that we dropped support for php 7.1 and added support for php 7.2, 7.3, 7.4 it is enough to note where we are.

after merging this, i will tag 1.2.0 to make it usable.

RSalo committed:
Add several changes

RSalo committed:
Add several changes

dbu commented on CHANGELOG.md:
```diff
@@ -1,7 +1,13 @@
 
+1.2.0
+-----
+* Increased minimum version PHP to 7.1.
```
```suggestion
* Increased minimum version PHP to 7.2.
```

RSalo replied:
Exactly!

RSalo committed:
Update CHANGELOG.md

Co-authored-by: David Buchmann <david@liip.ch>

RSalo committed:
Merge branch 'master' of https://github.com/RSalo/resource-rest-bundle

RSalo committed:
Merge branch 'master' of https://github.com/RSalo/resource-rest-bundle

RSalo commented:
1) Removed travis SYMFONY_VERSION
2) Included TEST_INSTALLATION env
3) Added CHANGELOG  1.2.0
4) Removed README requirements
5) Added COMPOSER_FLAGS minimum stability to PHP 7.2

@dbu did I do correct changes?

RSalo commented:
@dbu Is this normal that the lowest build crashed by 20 minutes timeout? https://travis-ci.com/github/RSalo/resource-rest-bundle/jobs/413947034

dbu commented on .travis.yml:
```diff
@@ -26,24 +22,16 @@ cache:
-    - php: 7.2
-      env: STABILITY=dev SYMFONY_VERSION=3.4.*
+    - PHP: 7.2
+      env: COMPOSER_FLAGS="--prefer-lowest"
```
the SYMFONY_VERSION constant helps composer to solve the versions because it reduces what it even looks at. maybe if you add that back it works. we probably also need the SYMFONY_DEPRECATIONS_HELPER=weak option to avoid failing because of deprecations (which we don't care about with lowest versions).

RSalo replied:
I would like to resolve the deprecations in a next PR...

dbu replied:
it is great if you can resolve deprecations that happen with normal builds. for the --prefer-lowest build, we should not try to solve them, it might even be impossible to solve them - i meant to add the deprecations helper constant only in the env of the --prefer-lowest

RSalo replied:
Ok. I got it and will do it.
There is the lower build failed with Prophecy exception. Any ideas on how to fix it?

dbu commented on .travis.yml:
```diff
@@ -52,8 +40,6 @@ matrix:
 before_install:
   - phpenv config-rm xdebug.ini || true
   - composer self-update
```
```suggestion
  - composer self-update --2
```
i suggest we start using composer 2. that should make the installations much faster according to reports of people who tried it.

RSalo committed:
Update .travis.yml

Co-authored-by: David Buchmann <david@liip.ch>

dbu commented on composer.json:
```diff
@@ -10,28 +10,29 @@
         "symfony/web-server-bundle": "^2.8 || ^3.3 || ^4.0",
-        "symfony/translation": "^2.8 || ^3.3 || ^4.0"
+        "symfony/translation": "^2.8 || ^3.3 || ^4.0",
+        "phpspec/prophecy": "~1.0"
```
as this is for dev only, i think there is no cost in forcing to a new version that works for sure. 

```suggestion
        "phpspec/prophecy": "^1.12"
```

RSalo replied:
Ah. I've just thought that is a stable version.

dbu replied:
it is, but maybe there are some new features that have been added meanwhile. --prefer-lowest applies to everything. for the dev-dependencies, we don't care about supporting a wide range - that is only valuable with the main dependencies, so that the bundle can be installed in as many environments as possible. (within reasons, which is why it was ok to drop php 7.1 and we eventually will want to drop symfony 2.8, likely when adding symfony 5 support)

RSalo committed:
Update composer.json

Co-authored-by: David Buchmann <david@liip.ch>

RSalo committed:
SYMFONY_DEPRECATIONS_HELPER

RSalo committed:
SYMFONY_DEPRECATIONS_HELPER

RSalo committed:
Merge branch 'master' of https://github.com/RSalo/resource-rest-bundle

dbu reviewed (approved):
yay, thanks a lot!

dbu commented:
https://github.com/symfony-cmf/resource-rest-bundle/releases/tag/1.2.0
